tests/CSharpTemp/CSharpTemp.Tests.cs

```diff
@@ -344,4 +345,12 @@ public void TestNullAttributedFunctionPtr()
         }
     }
 
+    [Test]
+    public unsafe void TestMarshallingOfCharPtr()
+    {
+        var obj = new CheckMarshllingOfCharPtr();
+        var str = obj.FuncRetCharPtr;
+        Assert.AreEqual('S', str[0]);
+        Assert.AreEqual('i', str[3]);
```

Comment:
I was getting 253/254 instead of 83 (for 'S') that was because I had done typo while checking return types to mention the return type marshalling. Now the tests have passed. Will push that and a few other changes tomorrow.